src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -77,40 +97,142 @@
           timeout-chan (http/json-response {:uuid uuid
                                             :processed false
                                             :timed_out true}
-                                           503)
+                                           http/status-unavailable)
           response-chan ([{:keys [command exception]}]
                          (let [base-response {:uuid uuid
                                               :processed true}]
                            (if exception
-                             (http/json-response (assoc base-response
-                                                        :timed_out false
-                                                        :error (str exception)
-                                                        :stack_trace (map str (.getStackTrace exception)))
-                                                 503)
+                             (http/json-response
+                              (assoc base-response
+                                     :timed_out false
+                                     :error (str exception)
+                                     :stack_trace (map str (.getStackTrace exception)))
+                              http/status-unavailable)
                              (http/json-response (assoc base-response
                                                         :timed_out false)
-                                                 200)))))))))
+                                                 http/status-ok)))))))))
+
+(def new-request-schema
+  {:params {(s/required-key "command") s/Str
+            (s/required-key "version") s/Str
+            (s/required-key "certname") s/Str
+            (s/required-key "received") s/Str
+            (s/optional-key "checksum") s/Str}
+   :body java.io.InputStream
+   s/Any s/Any})
+
+(defn-validated normalize-new-request
+  [{:keys [params body] :as req} :- new-request-schema]
+  (-> req
+      (update-in [:params "command"] str/replace "_" " ")
+      (update-in [:params "version"] #(Integer/parseInt %))))
+
+(def old-request-schema
+  (s/conditional
```

Comment:
A 1-armed conditional schema is a little funny... What happens if it's not a map?